Django 2.2 support

shurph opened the pull request:
These changes provide  an experimental Django 2.2 support.

And there are few fixes for #645 issue.

shurph commented:
@bogdanpetrea , could you please look at this when you  will have a chance?

bogdanpetrea commented on silver/tests/unit/test_pdf.py:
```diff
@@ -86,14 +86,14 @@ def test_pdf_mark_as_clean_mimic_2_threads():
-def test_pdf_mark_as_dirty_min_value():
-    # somehow inconsistent state
-    pdf = PDF.objects.create(dirty=-1)
+def test_pdf_mark_as_clean_min_value():
```
Why did this test have to be changed?

shurph replied:
The main purpose to fit to the new feature in Django 2.2 — extended constraints.
With 2.2 you can't apply `pdf = PDF.objects.create(dirty=-1)` because integrity error will be produced (`dirty` is PositiveIntegerField).

As I understand `test_pdf_mark_as_dirty_min_value` was written to be sure that `dirty` can not be below 0. So it seems like with the new 2.2 feature and also with you changes (https://github.com/silverapp/silver/commit/e4dd96ef44bf62f52a7f939a213af4e6bad1ed6a) the test_pdf_mark_as_clean_min_value is unnecessary, test_pdf_mark_as_clean_min_value is enough.

bogdanpetrea replied:
Makes sense. Thanks!

bogdanpetrea commented:
Also, you should add Django 2.2 to the CI builds:
https://github.com/silverapp/silver/blob/master/.drone.yml#L55
https://github.com/silverapp/silver/blob/master/.travis.yml#L11

shurph commented:
@bogdanpetrea I've added Django 2.2 to CI builds and also switched travis default image (Ubuntu 14.04) to the Ubuntu 16.04 because Django 2.2 requires a fresh SQLite version.

shurph commented:
@bogdanpetrea thank you for merging this PR. Couls you please release `0.8b2` with these changes?
